.github/workflows/mkdocs.yml

```
@@ -55,31 +58,30 @@ jobs:
           enable-commit-comment: false
           enable-commit-status: false
           overwrites-pull-request-comment: false
-          alias: ${{ steps.netlify-alias.outputs.alias }}
+          alias: ${{ steps.branch.outputs.name }}
           github-deployment-environment: Netlify
           github-deployment-description: Branch deployment preview
         env:
           NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_PERSONAL_ACCESS_TOKEN }}
           NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
 
-      # On push to master, deploy the docs to GitHub Pages
-      - name: Set up git author
+      # Deploy docs on push to master
+      - name: Set up environment variables
         if: github.ref == 'refs/heads/master'
         run: |
-          remote_repo="https://${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git"
-          git config --global user.name "${GITHUB_ACTOR}"
-          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
-          git remote rm origin
-          git remote add origin "${remote_repo}"
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          echo "MKDOCS_VERSION=$(mkdocs --version | cut -d " " -f 3)" >> $GITHUB_ENV
+          echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-9`" >> $GITHUB_ENV
+          echo "CUSTOM_DOMAIN=docs.pulse.codacy.com" >> $GITHUB_ENV
+          echo -e "User-agent: *\nSitemap: https://${CUSTOM_DOMAIN}/sitemap.xml" > "${GITHUB_WORKSPACE}/docs/robots.txt"
```

I confirmed on the GitHub documentation that the environment variables [only become available on subsequent steps](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).

So, the most straightforward and explicit way to create a `robots.txt` file that depends on the `CUSTOM_DOMAIN` environment variable is to have a dedicated step for this:

```suggestion

      - name: Create robots.txt
        if: github.ref == 'refs/heads/master'
        run: |
          echo -e "User-agent: *\nSitemap: https://${{ env.CUSTOM_DOMAIN }}/sitemap.xml" > "${GITHUB_WORKSPACE}/docs/robots.txt"
```

I'm going to apply the same change on codacy/docs, since right now the workflow fails to update the `robots.txt` file. :sweat_smile: